Integrate the latest changes in OTE tests using SC SDK

LeonidBeynenson opened the pull request.

Commit:
Update no_nncf_trace context

Commit:
Add CheckpointHookBeforeTraining

Commit:
Fix bug

Commit:
Remove unnecessary compression_ctrl.distributed() as DDP itself broadcasts all parameters

Commit:
Add is_in_nncf_tracing context; Clean up code

Commit:
Add CheckpointHookBeforeTraining to __all__;Remove extra lines

druzhkov-paul committed:
Merge pull request #128 from aleksei-kashapov/ote_with_updated_nncf

Update integration part with NNCF

druzhkov-paul committed:
Merge branch 'ote' into feature/dp/update_ov_wrapper

druzhkov-paul committed:
Merge pull request #102 from openvinotoolkit/feature/dp/update_ov_wrapper

OpenVINO wrapper update

druzhkov-paul committed:
Merge pull request #102 from openvinotoolkit/feature/dp/update_ov_wrapper

OpenVINO wrapper update

AlexanderDokuchaev committed:
Add support of alt_ssd_export for NNCF models

AlexanderDokuchaev committed:
fix name

asenina committed:
Fix wrapping outputsReplacement original forward for forward export.

druzhkov-paul committed:
Merge branch 'ote' into nncf_alt_ssd_export

asenina committed:
Add nncf init context manager.

AlexanderDokuchaev committed:
some fixes

AlexanderDokuchaev committed:
Merge branch 'nncf_alt_ssd_export' of github.com:AlexanderDokuchaev/mmdetection into nncf_alt_ssd_export

asenina committed:
Add pydocs.

LeonidBeynenson committed:
Add TODO in OTEDataset

LeonidBeynenson committed:
Add command line parameters to OTE sample

LeonidBeynenson committed:
Make change for consistency of total_epochs in the config

It is required for EpochBasedRunner.

LeonidBeynenson committed:
Move implementation of e2e_pytest decorator to a separate file

Ilya-Krylov committed:
fixed polygons visualization

Ilya-Krylov committed:
strong lexicons handling

Ilya-Krylov committed:
optimal metric value search

LeonidBeynenson committed:
Add the first version of ote training test; WIP

LeonidBeynenson committed:
Make the ote training test work

LeonidBeynenson committed:
Make parametrized ote training test

druzhkov-paul committed:
Merge pull request #135 from AlexanderDokuchaev/nncf_alt_ssd_export

Add support of alt_ssd_export for NNCF models

Ilya-Krylov committed:
Update mmdet/datasets/coco_with_text.py

Co-authored-by: Pavel Druzhkov <druzhkov.paul@gmail.com>

Ilya-Krylov committed:
cache lexicon

Ilya-Krylov committed:
minor

Ilya-Krylov committed:
Merge branch 'ote' into ik/some_things_for_text_spotting

Ilya-Krylov committed:
Merge pull request #137 from openvinotoolkit/ik/some_things_for_text_spotting

Ik/some things for text spotting

krzyczar committed:
Adding CollSys Manager

LeonidBeynenson committed:
Improve test_ote_training

LeonidBeynenson committed:
Merge pull request #138 from krzyczar/kcz/integrate_collsys_to_sc_sdk_tests

Adding CollSys Manager

Commit:
update sdk version

LeonidBeynenson committed:
Move the implementation of the class CollsysManager into a function

It is required to avoid bugs if e2e system is not installed.

Commit:
update custom models

Commit:
up

druzhkov-paul committed:
Merge pull request #136 from asenina/fix_bug_wrapping_output

Fix wrapping outputs. Replacement original forward for forward export.

Commit:
update models and tests

Ilya-Krylov committed:
fix?

Ilya-Krylov committed:
Merge branch 'ote' into det_out

LeonidBeynenson committed:
Add IMPT_COMMIT file

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/integrate_e2e_to_sc_sdk_tests_ver3

Ilya-Krylov committed:
Merge pull request #141 from openvinotoolkit/det_out

fix?

LeonidBeynenson committed:
Make test_ote_training work after merge, but not passed yet

Also make class in test_ote_training

LeonidBeynenson committed:
Fix bug to make config.evaluation.interval be positive even for small num iters

This make test_ote_training pass

LeonidBeynenson committed:
Make test_ote_training work correctly

LeonidBeynenson committed:
Fix constranints file to work with e2e==1.181.*

Commit:
update config manager

Ilya-Krylov committed:
nms_pre_classwise

Ilya-Krylov committed:
minor

LeonidBeynenson committed:
Make test_ote_training work with new DataCollector (aka TinySystem from e2e)

LeonidBeynenson committed:
Fix tests/user_config to allow connection to external server

Ilya-Krylov committed:
Merge pull request #144 from openvinotoolkit/feature/ik/nms_pre_classwise

nms_pre_classwise

LeonidBeynenson committed:
Make OTEDataset work without reading the whole dataset during initialization

LeonidBeynenson committed:
Add 'vitens_tiled' to test datasets

LeonidBeynenson committed:
Remove old comments

LeonidBeynenson committed:
Downgrade to python3.7 in init_venv2.sh

LeonidBeynenson committed:
Fix constraints.txt for OpenVINO 2021.4.582

Commit:
update sdk

Commit:
update sdk

Commit:
update hash

Commit:
clean up models' configs

LeonidBeynenson committed:
Merge main changes from the branch hlewando/integrate_e2e_to_sc_sdk

The merge is made manually, checking changes made
in hlewando/integrate_e2e_to_sc_sdk.

LeonidBeynenson committed:
Remove files unrequired in the future

Commit:
refactoring

LeonidBeynenson committed:
Merge branch 'hlewando/integrate_e2e_to_sc_sdk' into lb/integrate_e2e

Note that during the merge the changes in tests/test_ote_api.py were
removed to simplify with the later merges with Pavel's updates from the
branch dp/nous_sdk_api
(although some of the changes related to DataCollector were moved to
tests/test_ote_training.py)

LeonidBeynenson committed:
Add one more small change from hlewando/integrate_e2e_to_sc_sdk

LeonidBeynenson committed:
Merge branch 'lb/integrate_e2e_to_sc_sdk_tests_ver3' into lb/integrate_e2e

lukaszkaza commented on tests/test_ote_api.py:
```diff
@@ -27,8 +28,6 @@
 from mmdet.apis.ote.apis.detection import MMObjectDetectionTask, MMDetectionParameters, configurable_parameters
 
-from e2e_test_system import select_configurable_parameters
-from e2e.collection_system.systems import TinySystem
```
We would like to use TinySystem to measure performance of the tests.

lukaszkaza commented on tests/test_ote_api.py:
```diff
@@ -249,67 +230,25 @@ def train_and_eval(self, template_dir):
+        self.train_and_eval(osp.join('configs', 'ote', 'custom-object-detection', 'mobilenet_v2-2s_ssd-512x512'))
+
+    @e2e_pytest
+    def test_training_custom_mobilenet_atss(self):
```
There is no body of this function.

Commit:
add openvino task

LeonidBeynenson committed:
Make tests work with relative paths in config yaml files

LeonidBeynenson committed:
Merge branch 'lb/integrate_e2e_to_sc_sdk_tests_ver3' into lb/integrate_e2e

Add using relative paths in dataset_definitions.yml and
template_paths.yml.

Commit:
and export task to tests

LeonidBeynenson committed:
Fix bug in test_ote_training.py

The bug was in the function converting relative paths to absolute ones.

LeonidBeynenson committed:
Merge branch 'lb/integrate_e2e_to_sc_sdk_tests_ver3' into lb/integrate_e2e

Merge one bugfix.

Commit:
Merge branch 'ote' into dp/nous_sdk_api

LeonidBeynenson committed:
Fix bug -- removed one line by mistake

LeonidBeynenson committed:
Reverse some my changes before merging with the latest Pavel Druzhkov's changes

Note that it is done to make the merge more smooth.
At that this state of the project may be not fully functional at this
commit -- some tests or experiments may be failed.

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/integrate_e2e_to_sc_sdk_tests_ver3

Note that tests/test_ote_training.py is not updated after merge yet,
so it is not in work state.

LeonidBeynenson committed:
Remove IMPT_COMMIT file

Earlier this file was used to point the commit hash in impt repo that
should be used for SC SDK API.

Now the file mmdet/apis/ote/.sdk_sha.txt is used for that.

LeonidBeynenson committed:
Change to make work tests with very small total_iterations number

Without this change `config.evaluation.interval` was zero for such tests

LeonidBeynenson committed:
Make test_ote_training.py work

LeonidBeynenson committed:
Fix writing hyperparams to DataCollector in test_ote_training.py

LeonidBeynenson committed:
Add tests on export and evaluation of exported model to test_ote_training

LeonidBeynenson committed:
Merge branch 'lb/integrate_e2e_to_sc_sdk_tests_ver3' into lb/integrate_e2e

Merge updates -- new version of SC SDK, OpenVINO task, tests on export
and evaluation after export.

LeonidBeynenson committed:
Make test_ote_training work with real datasets and big number of iterations

LeonidBeynenson committed:
Merge branch 'lb/integrate_e2e_to_sc_sdk_tests_ver3' into lb/integrate_e2e

LeonidBeynenson committed:
Fix num checkpints in test_ote_training

LeonidBeynenson committed:
Merge branch 'lb/integrate_e2e_to_sc_sdk_tests_ver3' into lb/integrate_e2e

Very small update
